src/main/scala/com/gu/commercial/display/AdTargetParam.scala

```diff
@@ -0,0 +1,228 @@
+}
+
+case class ObserverParam(value: SingleValue) extends AdTargetParam {
+  override val name = ObserverParam.name
+}
+
+object ObserverParam {
+  val name = "ob"
+
+  def from(item: Content): Option[ObserverParam] =
+    SingleValue.fromRaw(if (item.tags.exists(_.id == "publication/theobserver")) "t" else "") map (ObserverParam(_))
+}
+
+case class PathParam(value: SingleValue) extends AdTargetParam {
+  override val name = PathParam.name
+}
+
+object PathParam {
+  val name = "url"
+
+  def apply(value: String): PathParam           = PathParam(SingleValue(value))
+  def fromItemId(id: String): Option[PathParam] = SingleValue.fromRaw(s"/${id.stripPrefix("/")}") map (PathParam(_))
+  def from(item: Content): Option[PathParam]    = fromItemId(item.id)
+  def from(section: Section): Option[PathParam] = fromItemId(section.id)
+  def from(tag: Tag): Option[PathParam]         = fromItemId(tag.id)
+}
+
+case class PlatformParam(value: SingleValue) extends AdTargetParam {
+  override val name = PlatformParam.name
+}
+
+object PlatformParam {
+  val name = "p"
+
+  def apply(value: String): PlatformParam = PlatformParam(SingleValue(value))
+}
+
+case class SeriesParam(value: MultipleValues) extends AdTargetParam {
+  override val name = SeriesParam.name
+}
+
+object SeriesParam {
+  val name = "se"
+
+  def from(item: Content): Option[SeriesParam] =
+    MultipleValues.fromTags(item) { tag =>
+      tag.`type` == Series || tag.paidContentType.contains("Series")
+    } map (SeriesParam(_))
+
+  def from(tag: Tag): Option[SeriesParam] =
+    if (tag.`type` == Series) MultipleValues.fromItemId(tag.id) map (SeriesParam(_))
+    else None
+}
+
+case class ShortUrlParam(value: SingleValue) extends AdTargetParam {
+  override val name = ShortUrlParam.name
+}
+
+object ShortUrlParam {
+  val name = "sh"
+
+  def from(item: Content): Option[ShortUrlParam] =
+    SingleValue.fromRaw(item.fields.flatMap(_.shortUrl).getOrElse("")) map (ShortUrlParam(_))
+}
+
+case class SurgeLevelParam(value: MultipleValues) extends AdTargetParam {
+  override val name = SurgeLevelParam.name
+}
+
+object SurgeLevelParam {
+  val name = "su"
+
+  def from(item: Content, surgeLookup: SurgeLookupService): Option[SurgeLevelParam] = {
+    val surgeLevels = bucket(surgeLookup.pageViewsPerMinute(item.id)).map(_.toString).toSet
+    MultipleValues.fromRawValues(surgeLevels) map (SurgeLevelParam(_))
+  }
+}
+
+case class ToneParam(value: MultipleValues) extends AdTargetParam {
+  override val name = ToneParam.name
+}
+
+object ToneParam {
+  val name = "tn"
+
+  def from(item: Content): Option[ToneParam] = MultipleValues.fromTagsOfType(item, Tone) map (ToneParam(_))
+
+  def from(tag: Tag): Option[ToneParam] =
+    if (tag.`type` == Tone) MultipleValues.fromItemId(tag.id) map (ToneParam(_))
+    else None
+}
+
+case object EmptyParam extends AdTargetParam {
+  override def name  = ""
+  override def value = SingleValue.empty
```

Comment:
Would this not be better called `UnknownParam`?

Just thought I would mention; this is a good reason why you would want an empty `sealed trait` for the ADT, because that means you can then have true types that represent emptiness without having to implement.
It's all a trade just off of boilerplate!